### PR TITLE
add lib_fuzzer and lib_fuzzer_no_main to llvm-project build

### PR DESCRIPTION
**Pull request by yakov-olkhovskiy**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Build libFuzzer libs from contrib/llvm-project

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Ok.

Interesting, why it is needed?
Usually, we link with compiler runtime libraries from the compiler itself - this is for sanitizers runtime and for fuzzer - at least it used to work for libFuzzer before.

**Comment by yakov-olkhovskiy** [comment]
@alexey-milovidov `-fsanitize=fuzzer` will attempt to link default libFuzzer library (usually installed with llvm) - which requires standard libc++ - as a result it will not link, since we are using custom built llvm libc++. With this addition we can now build fuzzers with `-fsanitize=fuzzer-no-link` flag, which enables instrumentation, and manually link this custom libFuzzer built against custom libc++

**Comment by nikitamikhaylov** [comment]
I'm wondering how it worked before

File: contrib/llvm-project-cmake/CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,39 @@
+set(COMPILER_RT_FUZZER_SRC_DIR "${ClickHouse_SOURCE_DIR}/contrib/llvm-project/compiler-rt/lib/fuzzer")
```
Comment (rschu1ze) [inline contrib/llvm-project-cmake/CMakeLists.txt:1]: 1. submodule "llvm-project" contains the entire LLVM repository. We currently only use it to build LLVM libraries for query compilation. Hopefully soon we will also build libcxx/libcxxabi from this submodule. But ideally, all usages reside in different directories in folder "contrib/". So my preference would be if you could move l. 1-36 into "contrib/libfuzzer-cmake/CMakeLists.txt".

2. Do we really want to build libfuzzer unconditionally? I know it is not very large but maybe there could be a CMake option `ENABLE_LIBFUZZER`. Iirc, there will be a special build for fuzzing in CI anyways.

**Review by rschu1ze — APPROVED** [review]

**Pull request merged**